### DISCUSSION TITLE
Any way to expose the functions executed on Promote Variation, Delete Move, Pring PGN and NAG Menu?

### DISCUSSION THREAD

**Discussion by coffeeDev98** [mliebelt/pgn-viewer#389] [Q&A]
Hey @mliebelt ,
Is there any way to expose the functions that are executed when the below UI is interacted with
<img width="318" alt="image" src="https://user-images.githubusercontent.com/63295908/186356952-7870b910-57d9-47df-9a00-efe2b28e259f.png">

I want to customize the UI but keep the functionality intact. So, Checking if I can create a UI outside the library but still be able to call the library functions.

**Comment by mliebelt** [comment]
I have to check, but of course, that seems to be reasonable. So you would build your own UI, but call the functions then for e.g. upvote a variation, print pgn, add comment, ...

**Reply by coffeeDev98** [reply]
Yes, ideally.

**Reply by coffeeDev98** [reply]
Right now, I'm able to manage this through selectors and using `click()` event

**Reply by mliebelt** [reply]
Which is ok (as workaround) but not ideal ...

**Reply by coffeeDev98** [reply]
Yes yes
